### PR TITLE
Fixed an issue that caused the basic auth prompt to fail in safari

### PR DESCRIPTION
**Pull request by m-kharbat**
## Summary of the changes / Why this improves CrateDB

From https://tools.ietf.org/html/rfc2617

> The realm directive (case-insensitive) is required for all
   authentication schemes that issue a challenge. The realm value
   (case-sensitive), in combination with the canonical root URL (the
   absoluteURI for the server whose abs_path is empty; see section 5.1.2
   of [2]) of the server being accessed, defines the protection space.
   These realms allow the protected resources on a server to be
   partitioned into a set of protection spaces, each with its own
   authentication scheme and/or authorization database. The realm value
   is a string, generally assigned by the origin server, which may have
   additional semantics specific to the authentication scheme. Note that
   there may be multiple challenges with the same auth-scheme but
   different realms.

The realm value was incorrectly used and caused basic authentication in safari to fail. 
related to https://github.com/crate/crate/pull/7121

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by m-kharbat** [comment]
It works when building it locally.. not sure how to test it otherwise

File: enterprise/users/src/main/java/io/crate/protocols/http/HttpAuthUpstreamHandler.java [hunk 1]
```diff
@@ -62,9 +62,7 @@
 
     private static final Logger LOGGER = LogManager.getLogger(HttpAuthUpstreamHandler.class);
     @VisibleForTesting
-    static final String WWW_AUTHENTICATE_REALM_MESSAGE = "Basic realm=\"Please provide credentials " +
-                                                         "(password maybe empty if trust authentication " +
-                                                         "is configured for your user)\")";
+    static final String WWW_AUTHENTICATE_REALM_MESSAGE = "Basic realm=\"CreateDB Authenticator\"";
```
Comment (mfussenegger) [inline enterprise/users/src/main/java/io/crate/protocols/http/HttpAuthUpstreamHandler.java]:
```suggestion
    static final String WWW_AUTHENTICATE_REALM_MESSAGE = "Basic realm=\"CrateDB Authenticator\"";
```

**Comment by m-kharbat** [comment]
@mfussenegger I think the parenthesis is what broke the realm previously. And it should not be used as a message but as an identifier of this user/password realm as explained in rfc2617

**Comment by m-kharbat** [comment]
@mfussenegger I didn't find anything in the spec to explain what characters are unsupported in the realm-value.. but safari fails to launch the password/user prompt when it contains special characters

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -82,3 +82,5 @@ Changes
 
 Fixes
 =====
+
+- Fixed an issue that caused the Basic Authentication prompt to fail in safari.
```
Comment (chaudum) [inline blackbox/docs/appendices/release-notes/unreleased.rst]:
```suggestion
- Fixed an issue that caused the Basic Authentication prompt to fail in Safari.
```

File: enterprise/users/src/main/java/io/crate/protocols/http/HttpAuthUpstreamHandler.java [hunk 3]
```diff
@@ -62,9 +62,9 @@
 
     private static final Logger LOGGER = LogManager.getLogger(HttpAuthUpstreamHandler.class);
     @VisibleForTesting
-    static final String WWW_AUTHENTICATE_REALM_MESSAGE = "Basic realm=\"Please provide credentials " +
-                                                         "(password maybe empty if trust authentication " +
-                                                         "is configured for your user)\")";
+
+    //realm-value should not contain any special characters
```
Comment (chaudum) [inline enterprise/users/src/main/java/io/crate/protocols/http/HttpAuthUpstreamHandler.java]:
```suggestion
    // realm-value should not contain any special characters
```

**Pull request merged**